### PR TITLE
api-connection

### PR DESCRIPTION
**Pull request by benjaglobant**
In this PR you can see the implementation of the classes needed to make the request to the API.
In the following image is shown the response content:
image

### CODE REVIEW

**Review by hernanhorquin96 — APPROVED** [review]

File: app/src/main/java/com/globant/weatherapp/activities/WeatherInfoActivity.kt [hunk 1]
```diff
@@ -21,7 +21,7 @@ class WeatherInfoActivity : AppCompatActivity() {
         val cityId = intent.extras.getSerializable(CITY_ID)
 
         presenter = WeatherPresenter(WeatherModel(), WeatherView(this))
-        presenter.initPresenter()
+        presenter.initPresenter(cityId.toString().toInt())
```
Comment (SartiMau) [inline app/src/main/java/com/globant/weatherapp/activities/WeatherInfoActivity.kt]: is this really necessary? can't you use only the .toInt() ?

File: app/src/main/java/com/globant/weatherapp/mvp/presenter/WeatherPresenter.kt [hunk 2]
```diff
@@ -1,13 +1,27 @@
...
 ) : WeatherContracts.Presenter {
 
-    override fun initPresenter() {
+    override fun initPresenter(cityId: Int) {
+        getFiveDaysWeather(cityId)
         view.initView()
     }
+
+    fun getFiveDaysWeather(cityId: Int) {
+        service.getFiveDaysWeatherByCityId(cityId)
```
Comment (SartiMau) [inline app/src/main/java/com/globant/weatherapp/mvp/presenter/WeatherPresenter.kt]: the presenter do not know the service.. that is the responsibility of the model.
You should call here model.getFiveDaysWeatherByCityId(cityId) and the model will call service.getFiveDaysWeatherByCityId(cityId).

**Review by BQuinteros — APPROVED** [review]

**Review by SartiMau — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/globant/weatherapp/activities/WeatherInfoActivity.kt (hunk 1 above)
Comment (benjaglobant) [reply app/src/main/java/com/globant/weatherapp/activities/WeatherInfoActivity.kt]: Refactored! Thanks Mau

File: app/src/main/java/com/globant/weatherapp/mvp/presenter/WeatherPresenter.kt (hunk 2 above)
Comment (benjaglobant) [reply app/src/main/java/com/globant/weatherapp/mvp/presenter/WeatherPresenter.kt]: Okey, done!

File: app/src/main/java/com/globant/weatherapp/mvp/model/WeatherModel.kt [hunk 3]
```diff
@@ -1,6 +1,19 @@
...
+import io.reactivex.schedulers.Schedulers
+
+class WeatherModel(private val service: WeatherService): WeatherContracts.Model {
+
+    override fun getFiveDaysWeather(cityId: Int) {
+        service.getFiveDaysWeatherByCityId(cityId)
+            .subscribeOn(Schedulers.io())
+            .observeOn(AndroidSchedulers.mainThread())
+            .subscribe({ forecast ->
+                val request = forecast
```
Comment (SartiMau) [inline app/src/main/java/com/globant/weatherapp/mvp/model/WeatherModel.kt:15]: what do you do with the service response?

Comment (benjaglobant) [reply app/src/main/java/com/globant/weatherapp/mvp/model/WeatherModel.kt:15]: In a future PR I will map the response and use the data to be shown in cardsView

**Review by SartiMau — APPROVED** [review]

**Review by lucianoeli — APPROVED** [review]

**Review by mauroLonghi — APPROVED** [review]

**Pull request merged**